src/main/java/reposense/RepoSense.java

```diff
@@ -50,9 +50,10 @@ public static void main(String[] args) {
             }
 
             RepoConfiguration.setFormatsToRepoConfigs(configs, cliArguments.getFormats());
-            RepoConfiguration.setDatesToRepoConfigs(configs, cliArguments.getSinceDate(), cliArguments.getUntilDate());
             ReportGenerator.generateReposReport(configs, cliArguments.getOutputFilePath().toAbsolutePath().toString(),
-                    formatter.format(ZonedDateTime.now(ZoneId.of("UTC+8"))));
+                    formatter.format(ZonedDateTime.now(ZoneId.of("UTC+8"))),
+                    cliArguments.getSinceDate().orElse(null),
```

Comment:
I believe the previous authors use Optional for some reason. Maybe due to avoid using null in our code. So I suggest you should replace all null with Optional in the code.

Reply:
In the original code the previous authors also resort to `orElse(null)`